setup.py

```diff
@@ -69,7 +69,7 @@
     license='MIT',
     scripts=[],
     data_files=[],
-    install_requires=["numpy>=1.16", "scipy>=1.6"],
+    install_requires=["numpy>=1.16", "scipy>=1.6", "setuptools>=60.0.0"],
```

Comment:
While it might seem strange to have `setuptools` as an install requirement, as it is needed at runtime this is the only way to ensure that it will be, as 

https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/pyproject.toml#L3

is only for build time not run time.

Reply:
There is an argument to be made that as the only use of 

https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/ot/helpers/openmp_helpers.py#L41

is in `setup.py`

https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/setup.py#L15-L16

https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/setup.py#L36

and so should be covered by the build system and only relevant at build time. However, it is then used in the tests:

https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/test/test_helpers.py#L26

This feels like a call that I think the developers should make, and so @rflamary @cedricvincentcuaz do you want:
* `setuptools` to be added alongside `pip` in all the `pip install`s of the tests
https://github.com/PythonOT/POT/blob/1f7d5476c48e04de32b6ec2a4ac94a72ba0ac2de/.github/workflows/build_tests.yml#L38
* `setuptools` is added as a runtime dependency

I can see arguments either way, but despite the PR status might slightly lean towards just adding the `pip install --upgrade setuptools` to the tests, as a user who isn't a developer should probably never need to touch `check_openmp_support`.